python/tank/util/authentication.py

```diff
@@ -0,0 +1,314 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Shotgun authentication for Toolkit
+"""
+
+import os
+import logging
+
+from tank_vendor.shotgun_api3 import Shotgun
+from tank_vendor.shotgun_api3 import AuthenticationFault
+from tank.errors import TankAuthenticationError
+from ConfigParser import SafeConfigParser
+from tank.util import shotgun
+from tank.util import path
+
+# Configure logging
+logger = logging.getLogger("sgtk.authentication")
+# logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler())
+
+_shotgun_instance_factory = Shotgun
+
+
+def _get_cached_login_info_location(base_url):
+    """
+    Returns the location of the session file on disk for a specific site.
+    :param base_url: The site we want the login information for.
+    :returns: Path to the login information.
+    """
+    from tank.util import path
+    return os.path.join(
+        path.get_local_site_cache_location(base_url),
+        "authentication",
+        "login.ini"
+    )
+
+
+def _is_session_token_cached():
+    """
+    Returns if there is a cached session token for the current user.
+    :returns: True is there is, False otherwise.
+    """
+    if get_cached_login_info(shotgun.get_associated_sg_config_data()["host"]):
+        return True
+    else:
+
+        return False
+
+
+def get_cached_login_info(base_url):
+    """
+    Returns the cached login info if found.
+    :returns: Returns a dictionnary with keys login and session_token or None
+    """
+    # Retrieve the location of the cached info
+    info_path = _get_cached_login_info_location(base_url)
+    # Nothing was cached, return an empty dictionary.
+    if not os.path.exists(info_path):
+        logger.debug("No cache found at %s", info_path)
+        return None
+    try:
+        # Read the login information
+        config = SafeConfigParser({"login": None, "session_token": None})
+        config.read(info_path)
+        if not config.has_section("LoginInfo"):
+            logger.debug("No Login info was found")
+            return None
+
+        login = config.get("LoginInfo", "login", raw=True)
+        session_token = config.get("LoginInfo", "session_token", raw=True)
+
+        if not login or not session_token:
+            logger.debug("Incomplete settings (login:%s, session_token:%s)", login, session_token)
```

Comment:
Normally, we use the syntax `logger.debug("Incomplete settings (login:%s, session_token:%s)" % (login, session_token)) - this generally seems more portable - for example you could replace `logger.debug` with `print` more easily.